src/Chapter4.hs

```diff
@@ -673,6 +729,21 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+data BTree a = BTree a (Maybe (BTree a)) (Maybe (BTree a))
```

Comment:
Interesting representation of the tree! 🌴 🎄 Great work!

The most common tree type in Haskell is:

```haskell
data Tree a
    = Empty
    | Node (Tree a) a (Tree a)
```

But your type is absolutely correct! 👍🏼

Reply:
Thanks, this is way more elegant, and it shows in the implementations
Had a go at that too :)